pkg/daemon/update.go

```diff
@@ -382,29 +382,42 @@ func (dn *CoreOSDaemon) applyOSChanges(mcDiff machineConfigDiff, oldConfig, newC
 		dn.nodeWriter.Eventf(corev1.EventTypeNormal, "OSUpdateStarted", mcDiff.osChangesString())
 	}
 
+	// We previously did not emit this event when kargs changed, so we still don't
 	if mcDiff.osUpdate || mcDiff.extensions || mcDiff.kernelType {
 		// We emitted this event before, so keep it
 		if dn.nodeWriter != nil {
 			dn.nodeWriter.Eventf(corev1.EventTypeNormal, "InClusterUpgrade", fmt.Sprintf("Updating from oscontainer %s", newConfig.Spec.OSImageURL))
 		}
-
 	}
 
-	// The steps from here on are different depending on the image type, so check the image type
-	isLayeredImage, err := dn.NodeUpdaterClient.IsBootableImage(newConfig.Spec.OSImageURL)
-	if err != nil {
-		return fmt.Errorf("Error checking type of update image: %w", err)
-	}
+	// Only check the image type and excute OS changes if:
+	// - machineconfig changed
```

Comment:
By this you mean the osImageURL? Looking below it seems that we don't want any MC change to trigger the OS update path, which I thought was the idea

Reply:
OSImageURL specifically, but also anything else in the mcDiff that would require us to extract something, since we do that differently based on the image format. 

Explaining this out loud kind of makes me think we need something like `mcdiff.RequresExtract` for right now and I should refactor applyOSChanges a little bit.

Reply:
Ok, I think I was generally confused by the sweeping `mc changed` and interpreted it as "files/units/sshkey", but that makes sense to me, thanks!

Reply:
As for splitting it out... probably worth doing but doesn't have to be part of this PR